subsys/net/lib/wifi_credentials/Kconfig

```diff
@@ -40,7 +40,7 @@ endchoice
 
 config WIFI_CREDENTIALS_MAX_ENTRIES
 	int "Number of supported WiFi credentials"
-	default 2
+	default 1
```

Comment:
> Hence reducing the max entries to 1 to avoid these 2 issues.

This sounds like a workaround than actual fix, what if the user configures this to 2?